compiler/fir/tree/tree-generator/src/org/jetbrains/kotlin/fir/tree/generator/ImplementationConfigurator.kt

```diff
@@ -469,6 +469,18 @@ object ImplementationConfigurator : AbstractFirTreeImplementationConfigurator()
             useTypes(expressionType)
         }
 
+        impl(resolvedQualifier) {
+            // Initialize the value to true if only the companion object is present. This makes standalone class reference expressions
+            // correctly resolve to the companion object. If it's a receiver expression of some other qualified access, the value is later
+            // corrected.
+            default("resolveToCompanionObject", "(symbol?.fir as? FirRegularClass)?.companionObject != null")
```

Comment:
So, does this flag record the fact that the qualifier is "resolved" to companion object or does it force some other part (e.g., resolver) to "resolve" this qualifier to companion object? To me, this default value is read like: hey, resolve this qualifier to the companion object if any.

If the former is the case (I think so), how about renaming this flag to `resolvedToCompanionObject`? Also, even for that standalone class reference, we can set this properly at `FirCallResolver#resolveCallableReference`. I.e., I'm concerning about this resolved-to-companion-obejct-if-present as a default value, e.g., what if it's read before RESOLVE state?

Reply:
It's the former case indeed. Let me do the renaming.

> Also, even for that standalone class reference, we can set this properly at FirCallResolver#resolveCallableReference. I.e., I'm concerning about this resolved-to-companion-obejct-if-present as a default value, e.g., what if it's read before RESOLVE state?

I am not sure I understand this part. This flag is updated in `FirCallResolver#resolveCallableReference` based on the resolved candidates. Is there something else that should be done here?

Reply:
My question is, if this flag is updated there based on the resolved candidates, what does this default value mean _before_ such update?

Reply:
As explained above, the default value makes standalone class reference resolved to companion object. For example
```
fun test() {
  val companionObjectOfClassFoo = Foo
}
```

Reply:
OK now I think about it again. There is no way it's gonna work if we remove this default value because `FirCallResolver#resolveCallableRefernce` won't be invoked for standalone class references. I am guessing your original question is due to ambiguous meaning of "standalone class reference" and you were probably thinking of `Foo.someMethod()` instead of `Foo` by itself.